plugin/kubernetes/controller.go

```diff
@@ -548,6 +558,9 @@ func endpointsSubsetDiffs(a, b *api.Endpoints) *api.Endpoints {
 
 // sendUpdates sends a notification to the server if a watch is enabled for the qname.
 func (dns *dnsControl) sendUpdates(oldObj, newObj interface{}) {
+	if len(dns.watched) == 0 {
+		return
+	}
```

Comment:
this would be good as a small separate PR. I've also have #2157 to disable/enable watching entire separately